### PR TITLE
Added imu full measured data publisher

### PR DESCRIPTION
**Pull request by VHolonec**

### CODE REVIEW

File: src/imu_full_measured_data_provider.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+void ImuFullMeasuredDataProvider::init()
+{
+    // initialize a library
+}
+
+imu_ros2::msg::ImuFullMeasuredData ImuFullMeasuredDataProvider::getData(bool& success)
+{
+    success = true;
```
Comment (rbolboac) [inline src/imu_full_measured_data_provider.cpp:40]: why do you have a success variable if you do not verify if the readings are successful? the get APIs below should return an error status and success should be updated according to their status

File: src/iio_wrapper.cpp [hunk 2]
```diff
@@ -427,6 +427,136 @@ float IIOWrapper::getTemperature()
...
+float IIOWrapper::getRawTemperature()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_temp, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleTemp / 1000.0;
+    return result;
+}
+
```
Comment (rbolboac) [inline src/iio_wrapper.cpp:559]: these APIs should return the status and the read value should be pass by pointer in the parameter list.
the name is confusing, please change to getRegTemperature, getRegVelocityZ, etc. you are not returning the raw value, you are returning the converted value by performing register reads.

File: msg/AdiImuData.msg [hunk 3]
```diff
@@ -9,11 +9,11 @@ geometry_msgs/Vector3 gyro
 # Internal Temperature [°C]
 float32 temp
 
-# delta velocity output [m/s2]
+# delta velocity output [m/s]
```
[the remark this answers is not in the document]
Comment (VHolonec) [reply msg/AdiImuData.msg]: I changed

File: msg/AdiImuData.msg [hunk 4]
```diff
@@ -9,11 +9,11 @@ geometry_msgs/Vector3 gyro
...
 
 geometry_msgs/Vector3 delta_vel
 
-# delta angle output [rad/s]
+# delta angle output [degrees]
```
[the remark this answers is not in the document]
Comment (VHolonec) [reply msg/AdiImuData.msg]: I changed

File: src/imu_full_measured_data_ros_publisher.cpp [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    timenow done;
+    bool success = false;
+
+    rclcpp::WallRate loopRate(0.1);
+
+    while (rclcpp::ok()) {
+
+       //std::thread::id this_id = std::this_thread::get_id();
+        //std::cout << "thread " << this_id << " running...\n";
+        //RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "running: '%d'", this_id);
```
[the remark this answers is not in the document]
Comment (VHolonec) [reply src/imu_full_measured_data_ros_publisher.cpp]: ok

File: msg/AdiImuData.msg (hunk 4 above)
[the remark this answers is not in the document]
Comment (rbolboac) [reply msg/AdiImuData.msg]: you did not have to remove these changes altogether. should just add them in another commit, can be done in the same PR

File: msg/AdiImuData.msg (hunk 3 above)
[the remark this answers is not in the document]
Comment (rbolboac) [reply msg/AdiImuData.msg]: you did not have to remove these changes altogether. should just add them in another commit, can be done in the same PR

File: src/imu_full_measured_data_provider.cpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+    success = m_iioWrapper.getRegRotZ(data.delta_angle.z);
+    if(!success)
+        return false;
+
+    success = m_iioWrapper.getRegTemperature(data.temp);
+    if(!success)
+        return false;
+
+    return true;
```
Comment (rbolboac) [inline src/imu_full_measured_data_provider.cpp:94]: can be replaced with a single line: return success

File: src/imu_full_measured_data_ros_publisher.cpp [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu gyroscope data: '%f' '%f' '%f'",
+                        m_message.gyro.x, m_message.gyro.y, m_message.gyro.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu delta velocity data: '%f' '%f' '%f'",
+                        m_message.delta_vel.x, m_message.delta_vel.y, m_message.delta_vel.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu delta angle data: '%f' '%f' '%f'",
+                        m_message.delta_angle.x, m_message.delta_angle.y, m_message.delta_angle.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu temperature data: '%f' ",
+                        m_message.temp);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu time %d us ",
+                        std::chrono::duration_cast<std::chrono::microseconds>(done-started).count());
```
Comment (rbolboac) [inline src/imu_full_measured_data_ros_publisher.cpp:80]: I don't think this data should be logged always. if success is false we should log that success is false and read data is not valid. is success is true we can log the values

File: src/imu_full_measured_data_provider.cpp [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+void ImuFullMeasuredDataProvider::init()
+{
+    // initialize a library
+}
+
+bool ImuFullMeasuredDataProvider::getData(imu_ros2::msg::ImuFullMeasuredData& data)
+{
+    bool success = true;
```
Comment (rbolboac) [inline src/imu_full_measured_data_provider.cpp:40]: nit: in this case success does not need to be initialized

**Review by rbudai98 — APPROVED** [review]

**Review by SeptimiuVana — APPROVED** [review]

**Pull request merged**